### PR TITLE
Use admin URL for time mismatch warning in Jetpack env

### PR DESCRIPTION
**Pull request by ChaosExAnima**
#### Changes proposed in this Pull Request

* Creates a new selector, `getSettingsUrl`, which returns either a Calypso or wp-admin path based on environment.
* Wires up selector to `TimeMismatchWarning` component to send users to correct path in all envs.

Note: right now the selector only handles the general settings, as there's no map between wp-admin and Calypso settings paths that I could leverage.

#### Testing instructions

1. Set up a JP site with a timezone not in yours.
2. Check activity log, backup, and scan pages in Calypso environment.
3. Verify notice appears and link takes you to the Calypso settings page.
4. Check again, this time with Jetpack environment.
5. Verify notice appears and link takes you to the wp-admin settings page.

### CODE REVIEW

**Review by rcanepa — CHANGES_REQUESTED** [review]
Works great in the cloud but in WordPress.com it doesn't work. There is no link that takes you to the Settings section.

File: client/state/selectors/get-settings-url.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	const siteIsWPCOM = isSiteWPCOM( state, siteId );
+	if ( isJetpackCloud() && ! siteIsWPCOM ) {
+		if ( section === 'general' ) {
+			return `${ siteAdminUrl }options-general.php`;
+		}
+	} else if ( siteIsWPCOM ) {
+		if ( section === 'general' ) {
+			return `/settings/general/${ siteSlug }`;
+		}
+	}
```
Comment (rcanepa) [inline client/state/selectors/get-settings-url.ts:34]: I think you missed the Jetpack site in WordPress.com case.

**Comment by rcanepa** [comment]
Not sure if this related to this PR, but the X to dismiss de Notice doesn't seem to do anything.

File: client/state/selectors/get-settings-url.ts (hunk 1 above)
Comment (ChaosExAnima) [reply client/state/selectors/get-settings-url.ts:34]: Oh, that'd make sense wouldn't it? 😅

**Comment by ChaosExAnima** [comment]
> Not sure if this related to this PR, but the X to dismiss de Notice doesn't seem to do anything.

Good catch! I'm honestly unsure how this should function- should it always dismiss? Dismiss per site? Per timezone? It's out of scope for this version, but it's a good thing to explore.

**Comment by rcanepa** [comment]
> > Not sure if this related to this PR, but the X to dismiss de Notice doesn't seem to do anything.
> 
> Good catch! I'm honestly unsure how this should function- should it always dismiss? Dismiss per site? Per timezone? It's out of scope for this version, but it's a good thing to explore.

I agree! We should address this in a different PR once we understand what the dismiss button should do.

File: client/state/selectors/get-settings-url.ts (hunk 1 above)
Comment (ChaosExAnima) [reply client/state/selectors/get-settings-url.ts:30]: Resolved in b98d32a. Can you retest?

**Review by rcanepa — APPROVED** [review]
It works great now in both Calypsos!

**Pull request merged**